ssl/t1_lib.c

```diff
@@ -3170,10 +3170,11 @@ static int tls_decrypt_ticket(SSL *s, const unsigned char *etick,
     p = sdec;
 
     sess = d2i_SSL_SESSION(NULL, &p, slen);
+    slen -= p - sdec;
```

Comment:
Generally speaking, if I am forced to use type casts in this way, it is an indication
that I have a problem with the interfaces;

For instance

slen is int,
d2i_SSL_SESSION uses long,
and p-sdec is size_t
I would have no need for type casts if all of them would use size_t